x-pack/platform/plugins/shared/onechat/public/application/components/connector_selector/connector_selector.tsx

```diff
@@ -0,0 +1,117 @@
+  type ConnectorSelectableComponentProps,
+} from '@kbn/ai-assistant-connector-selector-action';
+import type { InferenceConnector } from '@kbn/inference-common';
+import { InferenceConnectorType } from '@kbn/inference-common';
+import { useNavigation } from '../../hooks/use_navigation';
+
+interface ConnectorSelectorProps {
+  connectors: InferenceConnector[];
+  selectedConnectorId?: string;
+  onSelectConnector: (connectorId: string) => void;
+  isLoading?: boolean;
+  defaultConnectorId?: string;
+}
+
+export const ConnectorSelector: React.FC<ConnectorSelectorProps> = ({
+  connectors,
+  selectedConnectorId,
+  onSelectConnector,
+  isLoading = false,
+  defaultConnectorId,
+}) => {
+  const { euiTheme } = useEuiTheme();
+  const { navigateToManageConnectors } = useNavigation();
+  const [isPopoverOpen, setIsPopoverOpen] = React.useState(false);
+
+  const togglePopover = () => setIsPopoverOpen(!isPopoverOpen);
+  const closePopover = () => setIsPopoverOpen(false);
+
+  const panelStyles = css`
+    inline-size: calc(${euiTheme.size.xxl} * 8);
+  `;
+
+  // Split connectors into preconfigured and custom
+  const { preConfiguredConnectors, customConnectors } = useMemo(() => {
+    const preConfigured: ConnectorSelectableComponentProps['preConfiguredConnectors'] = [];
+    const custom: ConnectorSelectableComponentProps['customConnectors'] = [];
+
+    connectors.forEach((connector) => {
+      const option = {
+        value: connector.connectorId,
+        label: connector.name || connector.connectorId,
+      };
+
+      // Consider inference type connectors as "preconfigured"
+      if (connector.type === InferenceConnectorType.Inference) {
+        preConfigured.push(option);
+      } else {
+        custom.push(option);
+      }
+    });
+
+    return { preConfiguredConnectors: preConfigured, customConnectors: custom };
+  }, [connectors]);
+
+  const selectedConnector = connectors.find((c) => c.connectorId === selectedConnectorId);
+  const selectedConnectorName = selectedConnector?.name || selectedConnector?.connectorId;
+
+  if (isLoading) {
+    return <EuiLoadingSpinner size="m" />;
+  }
+
+  const button = (
+    <EuiButtonEmpty
+      iconType="arrowDown"
+      iconSide="right"
+      onClick={togglePopover}
+      disabled={connectors.length === 0}
+      data-test-subj="onechatConnectorSelectorButton"
+      aria-haspopup="menu"
+      aria-label={i18n.translate('xpack.onechat.connectorSelector.selectConnector', {
+        defaultMessage: 'Select connector',
+      })}
+    >
+      {selectedConnectorName ||
+        i18n.translate('xpack.onechat.connectorSelector.noConnector', {
+          defaultMessage: 'No connector',
+        })}
+    </EuiButtonEmpty>
+  );
+
+  return (
+    <EuiPopover
+      panelProps={{ css: panelStyles }}
+      button={button}
+      isOpen={isPopoverOpen}
+      closePopover={closePopover}
+      panelPaddingSize="none"
+      anchorPosition="upLeft"
+    >
+      <ConnectorSelectable
```

Comment:
Not related to the PR or what you did, but I was somewhat hoping that this component would be slighly "higher" level. We have to load all the data and everything here, which feels wrong in terms of code duplication...

Reply:
That was a very good point, I simplified by using a common hook in https://github.com/elastic/kibana/pull/238476/commits/c0bca14eb3f31cc8345868ff5ee288288b60964f